pkg/service/service.go

```diff
@@ -1,13 +1,125 @@
 package service
 
 import (
+	"context"
+
 	runapi "github.com/GoogleCloudPlatform/cloud-run-release-operator/internal/run"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/pkg/config"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/run/v1"
 )
 
-// Client is the connection to update and obtain data about a single service.
+// Client is the connection to update and obtain data about a single service. It
```

Comment:
That's possible. However, just take into consideration that we probably want to run the rollout process concurrently, so we need a `service.Client` per filtered service.

That being said, I would probably need to create a cache of `run.Client`s for regions the filter has seen already. Is it worth doing so in this PR or should we do it in another one?

Reply:
After discussing it with Ahmet, I think there's no harm in creating as many `run.Client`s as needed since initializing them does not make any HTTP requests per sé. So, there's no connection that should be closed or anything like that, and it's not using much memory. Therefore, I think it's better to creating the object as needed to increase code readability.